### DISCUSSION TITLE
Unicodeの絵文字とTwemojiが混在できる

### DISCUSSION THREAD

**Discussion by munierujp** [increments/qiita-discussions#467] [要望・アイデア・不具合]
Qiitaでは、`:sushi:`のように`:`で囲むことで絵文字（Twemoji）を埋め込める機能があります。

https://qiita.com/Qiita/items/c686397e4a0f4f11683d#%E7%B5%B5%E6%96%87%E5%AD%97

そこで、次のようなMarkdownを書くと……

```md
||Unicode|Qiita|
|---|---|---|
|寿司|🍣|:sushi:|
|ビール|🍺|:beer:|
```

このようなHTMLに変換されます。

```html
<table data-sourcepos="1:1-4:23">
<thead>
<tr data-sourcepos="1:1-1:16">
<th data-sourcepos="1:2-1:1"></th>
<th data-sourcepos="1:3-1:9">Unicode</th>
<th data-sourcepos="1:11-1:15">Qiita</th>
</tr>
</thead>
<tbody>
<tr data-sourcepos="3:1-3:21">
<td data-sourcepos="3:2-3:7">寿司</td>
<td data-sourcepos="3:9-3:12">🍣</td>
<td data-sourcepos="3:14-3:20"><img alt=":sushi:" class="emoji" height="20" src="https://cdn.qiita.com/emoji/twemoji/unicode/1f363.png" title=":sushi:" width="20"></td>
</tr>
<tr data-sourcepos="4:1-4:23">
<td data-sourcepos="4:2-4:10">ビール</td>
<td data-sourcepos="4:12-4:15">🍺</td>
<td data-sourcepos="4:17-4:22"><img alt=":beer:" class="emoji" height="20" src="https://cdn.qiita.com/emoji/twemoji/unicode/1f37a.png" title=":beer:" width="20"></td>
</tr>
</tbody>
</table>
```

これは、このように表示されます。

<img width="220" alt="スクリーンショット 2023-04-23 21 43 08" src="https://user-images.githubusercontent.com/20086673/233840510-0ca720e7-ab49-4449-bcc6-1ea34ef9e5a7.png">

Unicodeの絵文字とTwemojiが混在できるようになっていますが、これは意図した仕様でしょうか？
たとえばTwitterでは、Unicodeの絵文字を投稿しようとしてもツイート画面に入力した段階でTwemojiに変換されます。

**Comment by tanaka-asami415** [comment]
@munierujp 
>Unicodeの絵文字とTwemojiが混在できるようになっていますが、これは意図した仕様でしょうか？

現時点では、UnicodeとTwemojiの絵文字をそれぞれ別に扱っており、
変換処理をしていないため、意図した仕様となります🙇

**Reply by munierujp** [reply]
意図した仕様ということで理解しました。ありがとうございます。
